sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplification.cs

```diff
@@ -119,6 +119,11 @@ private static bool AreTypesCompatible(ExpressionSyntax expression1, ExpressionS
             var type1 = semanticModel.GetTypeInfo(expression1).Type;
             var type2 = semanticModel.GetTypeInfo(expression2).Type;
 
+            if (expression1 is AnonymousFunctionExpressionSyntax || expression2 is AnonymousFunctionExpressionSyntax)

```

Comment:
Minor optimisation: this check can be done right at the start of the method, before the calls to get `type1` and `type2` (rule of thumb is to do syntax checks first before touching the semantic model since fetching from the semantic model is always slower)